docstore/driver/document.go

```diff
@@ -150,6 +150,21 @@ func (d Document) SetField(field string, value interface{}) error {
 	return nil
 }
 
+// FieldNames returns names of the top-level fields of d.
+func (d Document) FieldNames() []string {
```

Comment:
Can you just use the fieldsCache?

Reply:
Can you sketch out what the code would be? I'm not sure I understand.

Reply:
nvm, I didn't see `Document.fields` contains only struct fields, I thought it was both.